driver-kotlin-coroutine/build.gradle.kts

```diff
@@ -0,0 +1,189 @@
+
+    integrationTestImplementation("org.jetbrains.kotlin:kotlin-test-junit")
+    integrationTestImplementation(project(path = ":driver-sync"))
+    integrationTestImplementation(project(path = ":driver-core"))
+    integrationTestImplementation(project(path = ":bson"))
+}
+
+kotlin { explicitApi() }
+
+tasks.withType<KotlinCompile> { kotlinOptions.jvmTarget = "1.8" }
+
+// ===========================
+//     Code Quality checks
+// ===========================
+spotless {
+    kotlinGradle {
+        ktfmt("0.39").dropboxStyle().configure { it.setMaxWidth(120) }
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+        licenseHeaderFile(rootProject.file("config/mongodb.license"), "(group|plugins|import|buildscript|rootProject)")
+    }
+
+    kotlin {
+        target("**/*.kt")
+        ktfmt().dropboxStyle().configure { it.setMaxWidth(120) }
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+        licenseHeaderFile(rootProject.file("config/mongodb.license"))
+    }
+
+    format("extraneous") {
+        target("*.xml", "*.yml", "*.md")
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+    }
+}
+
+tasks.named("check") { dependsOn("spotlessApply") }
+
+detekt {
+    allRules = true // fail build on any finding
+    buildUponDefaultConfig = true // preconfigure defaults
+    config = rootProject.files("config/detekt/detekt.yml") // point to your custom config defining rules to run,
+    // overwriting default behavior
+    baseline = rootProject.file("config/detekt/baseline.xml") // a way of suppressing issues before introducing detekt
+    source =
+        files(
+            file("src/main/kotlin"),
+            file("src/test/kotlin"),
+            file("src/integrationTest/kotlin"),
+        )
+}
+
+tasks.withType<Detekt>().configureEach {
+    reports {
+        html.required.set(true) // observe findings in your browser with structure and code snippets
+        xml.required.set(true) // checkstyle like format mainly for integrations like Jenkins
+        txt.required.set(false) // similar to the console output, contains issue signature to manually edit
+    }
+}
+
+spotbugs {
+    showProgress.set(true)
+
+    tasks.getByName("spotbugsIntegrationTest") { enabled = false }
+}
+
+// ===========================
+//     Test Configuration
+// ===========================
+val integrationTest =
+    tasks.create("integrationTest", Test::class) {
+        description = "Runs the integration tests."
+        group = "verification"
+
+        testClassesDirs = sourceSets["integrationTest"].output.classesDirs
+        classpath = sourceSets["integrationTest"].runtimeClasspath
+    }
+
+tasks.create("kCheck") {
+    description = "Runs all the kotlin checks"
+    group = "verification"
+
+    dependsOn("clean", "check", integrationTest)
+    tasks.findByName("check")?.mustRunAfter("clean")
```

Comment:
This is just to ensure ordering for the depended on items.  

Makes sense to clean first then check, rather than check then clean. I think it caused an issue and prompted me to set the order explicitly.